subprojects/packagefiles/qarchive/meson.build

```diff
@@ -0,0 +1,69 @@
+project(
+  'QArchive',
+  'cpp',
+  version: '2.1.1',
+  license: 'BSD-3-Clause',
+  default_options: ['cpp_std=c++11'],
+  meson_version: '>=0.59.0') # because of compile_moc
```

Comment:
Hmm? I remember @eli-schwartz complaining about this before.

Reply:
Right. It's possible to support versions of Meson older than 0.59.0 by using `qt_mod.preprocess` instead of the `compile_moc` method.

There are plenty of people working on machines that have a system Meson older than 0.59.0, supporting those is a nice-to-have and doesn't really cost anything.

Reply:
Is there a minimum version you're targeting? Looks like `qt5.preprocess()` will work too and is older.

Reply:
There's no hard and fast rule. qt5.preprocess has existed since Meson 0.22, i.e. since the stone age, so it's going to work basically regardless of version.

https://repology.org/project/meson/versions lists the accessibility of Meson in various places. Of particular note is:
- Debian oldstable, has Meson 0.49.2
- Debian stable, has Meson 0.56.2
- Ubuntu 20.04 LTS, has Meson 0.53.2

The more places it works, the better. On the flip side, sometimes you really, really need features from the latest version of Meson.

Where to find the balance is a tough question, and the answer tends to be different depending on the project you ask. Since the main point of the WrapDB is to be consumed by other projects, I like to be conservative about it, and be compatible with as many versions as I can, unless it subjectively feels too hard or complex to support. Because we don't know what the compat policy is of every project that wants to install this wrap file.

In the case of "use one older function instead of a newer function", it seems pretty easy to support.

Reply:
Note that even without qt.compile_moc, you still use the following:
```
qarchive| * 0.47.0: {'dict'}
qarchive| * 0.49.0: {'configuration_data dictionary'}
qarchive| * 0.54.0: {'meson.override_dependency'}
```

override_dependency is not really crucial, since the `[provide]` section also ensures mapping the dependency, although use of the override does perform a consistency check.
- You could drop it.
- You could guard it inside `if meson.version().version_compare('>=0.54.0')`
- You could just require `meson_version: '>=0.54.0'`

Attempting to support Meson less than 0.49.0 is probably a waste of time.